container_pipeline/trigger_dockerfile_lint.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python
+
+import json
+import os
+
+from container_pipeline.vendors import beanstalkc
+# from container_pipeline.pipeline import create_project
+import utils
+
+
+# if __name__ == "__main__":
+# def trigger_dockerfile_linter(namespace, git_path, filename, notify_email,
+#                               job_name, logs_dir, **kwargs):
+def trigger_dockerfile_linter(appid, jobid, repo_url, repo_branch,
+                              repo_build_path, target_file, notify_email,
+                              desired_tag, depends_on, test_tag):
+    bs = beanstalkc.Connection(host="BEANSTALK_SERVER")
+    bs.use("master_tube")
+
+    job_name = utils.get_job_name({
+        'appid': appid, 'jobid': jobid, 'desired_tag': desired_tag})
+    logs_dir = '/srv/pipeline-logs/{}'.format(test_tag)
+
+    # namespace = sys.argv[1]
+    # git_path = sys.argv[2]
+    # filename = sys.argv[3]
+    # notify_email = sys.argv[4]
+    # job_name = sys.argv[5]
+    # logs_dir = sys.argv[6]
+
+    # having '/' in a value used in os.path.join generates unexpected paths
+    if repo_build_path.startswith("/"):
+        git_path = repo_build_path[1:]
+
+    if target_file.startswith("/"):
+        filename = target_file[1:]
+    else:
+        filename = target_file
+
+    lint_job_data = {}
+
+    try:
+        dockerfile_location = \
+            os.path.join(os.environ['DOCKERFILE_DIR'], git_path, filename)
+        with open(dockerfile_location) as f:
+            dockerfile = f.read()
+    except IOError as e:
+        print "==> Error opening the file %s" % dockerfile_location
+        print "==> Error: %s" % str(e)
+        print "==> Sending Dockerfile linter failure email"
+        response = {
+            "lint_status": True,
+            "action": "notify_user",
+            "namespace": appid,
+            "notify_email": notify_email,
+            "job_name": job_name,
+            "msg": "Couldn't find the Dockerfile at specified git_path",
+            "logs_dir": logs_dir,
+            "lint_status": False,
+            "build_status": False
+        }
+
+        jid = bs.put(json.dumps(response))
+        print "==>Put job on 'master_tube' tube with id: %d" % jid
+    else:
+        lint_job_data["appid"] = appid
+        lint_job_data["job_name"] = job_name
+        lint_job_data["dockerfile"] = dockerfile
+        lint_job_data["notify_email"] = notify_email
+        lint_job_data["logs_dir"] = logs_dir
+        lint_job_data["action"] = "start_linter"
+        lint_job_data["jobid"] = jobid
+        lint_job_data["repo_url"] = repo_url
+        lint_job_data["repo_branch"] = repo_branch
+        lint_job_data["repo_build_path"] = repo_build_path
+        lint_job_data["target_file"] = target_file
+        lint_job_data["desired_tag"] = desired_tag
+        lint_job_data["depends_on"] = depends_on
+        lint_job_data["test_tag"] = test_tag
+
+        bs.put(json.dumps(lint_job_data))
```

Comment:
dont we want use lib.queue here? so that we can handle beanstalk connections in one place.